src/current/v24.3/wal-failover.md

```diff
@@ -0,0 +1,495 @@
+---
+title: WAL Failover
+summary: Mitigate the effects of disk stalls in cloud deployments by failing over WAL writes to a secondary store.
+toc: true
+docs_area: deploy
+---
+
+{% include {{ page.version.version }}/wal-failover-intro.md %}
+
+This page has detailed information about WAL failover, including:
+
+- How WAL failover works.
+- How to enable WAL failover.
+- How to disable WAL failover.
+- How to test WAL failover.
+- How to monitor WAL failover.
+- How to configure WAL failover in multi-store configurations.
+- Frequently Asked Questions about WAL failover.
+
+For basic information about WAL failover, see [`cockroach start` > WAL failover]({% link {{ page.version.version }}/cockroach-start.md %}#write-ahead-log-wal-failover).
+
+## Why WAL Failover?
+
+In cloud environments transient [disk stalls]({% link {{ page.version.version }}/cluster-setup-troubleshooting.md %}#disk-stalls) are common, often lasting on the order of several seconds. This will negatively impact latency for the user-facing foreground workload. In the field, we have observed that stalls occur most frequently while writing to the WAL. While we cannot prevent disk stalls, we can minimize their impact on CockroachDB. That is where WAL failover comes into play.
+
+When a disk stalls on a node, it could be due to complete hardware failure or it could be a transient stall. In CockroachDB when a disk backing a [store]({% link {{ page.version.version}}/cockroach-start.md %}#store) stalls, all the writes to [ranges]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-range) for which the node is [leaseholder]({% link {{ page.version.version }}/architecture/overview.md %}#architecture-leaseholder) will be blocked until the disk stall clears, or the node is crashed (after default interval defined by [`COCKROACH_ENGINE_MAX_SYNC_DURATION_DEFAULT`](#important-environment-variables)), moving any leaseholders on this store to other stores.
+
+WAL failover uses a secondary disk to failover WAL writes to when transient disk stalls occur. This limits the write impact to a few hundreds of milliseconds (the [failover threshold, which is configurable]()). Note that WAL failover _only preserves availability of writes_. If reads to the underlying storage are also stalled, operations that read and do not find data in the block cache or page cache will stall.
+
+## Creating and configuring a cluster to be ready for WAL Failover
+
+The steps to provision a cluster that has single data store vs a multi-store cluster are slightly different. In this section, we will provide high-level instructions for setting up each of these configurations. We will use [GCE](https://cloud.google.com/compute/docs) as the environment. You will need to translate these instructions into the steps used by the deployment tools in your environment.
+
+<a name="single-store-config"></a>
+
+### Provisioning a cluster with single data store, single failover store for WAL failover
```

Comment:
nit: maybe we should say "single failover disk" or "single failover volume" to avoid using the word "store," since a store is its own concept in the CockroachDB ecosystem.

Reply:
updated to "single failover disk" since we also use that term elsewhere in the document

thanks for the catch and the reminder about our "store" terminology